### PR TITLE
Add datasets loading to cmake

### PR DESCRIPTION
**Pull request by TolyaTalamanov**

### CODE REVIEW

**Comment by TolyaTalamanov** [comment]
From CI log:
```
$ ./samples/mnist/run_mnist_train.sh
Epoch: 0
Loss: 0.61095
Loss: 0.448612
Loss: 0.371187
Loss: 0.328572
Loss: 0.295607
Loss: 0.270212
Loss: 0.252053
Loss: 0.244562
Accuracy: 0.96234
Model saved: build/mnist_model.bin
The command "./samples/mnist/run_mnist_train.sh" exited with 0.
0.27s$ ./samples/mnist/run_mnist_test.sh
Accuracy: 0.96234
The command "./samples/mnist/run_mnist_test.sh" exited with 0.
```

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,11 +19,34 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
...
-option(BUILD_SAMPLES   "Build samples"                        ON)
-option(BUILD_BENCHMARKS "Build benchmarks"                    ON)
-
+option(BUILD_TESTS      "Build deepworks with tests"         ON)
+option(WITH_EIGEN       "Build deepworks with eigen backend" ON)
+option(BUILD_SAMPLES    "Build samples"                      ON)
+option(BUILD_BENCHMARKS "Build benchmarks"                   ON)
+option(DOWNLOAD_DATA    "Download and unpack datasets"       ON)
+
+if (DOWNLOAD_DATA)
```
Comment (TolyaTalamanov) [inline CMakeLists.txt:28]: Possibly need to move to separate file

File: samples/mnist/run_mnist_test.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+if test -z ${DW_BUILD_PATH}
```
Comment (TolyaTalamanov) [inline samples/mnist/run_mnist_test.sh:1]: Put this check to `test-env.sh` which will be enable for all bash tests

**Review by TolyaTalamanov — COMMENTED** [review]
Appropriate README.md is required

**Comment by TolyaTalamanov** [comment]
@apertovs Have a look, please

**Comment by TolyaTalamanov** [comment]
```
========================================================================================
||                    ||     Deepworks      ||       Torch        ||    DW vs Torch     ||
========================================================================================
||       Epochs       ||         10         ||         10         ||        None        ||
||     Train time     ||      26496 ms      ||      32846 ms      ||      -23.97%       ||
||   Inference time   ||      2852 ms       ||      2036 ms       ||      +40.08%       ||
||      Accuracy      ||      0.974800      ||      0.974059      ||       +0.08%       ||
||        Loss        ||      0.069351      ||      0.058649      ||      +18.25%       ||
========================================================================================
```

**Review by apertovs — APPROVED** [review]

File: CMakeLists.txt [hunk 3]
```diff
@@ -19,18 +19,22 @@ set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
...
-option(BUILD_TESTS     "Build deepworks with tests"           ON)
-option(WITH_EIGEN      "Build deepworks with eigen backend"   ON)
-option(BUILD_SAMPLES   "Build samples"                        ON)
-option(BUILD_BENCHMARKS "Build benchmarks"                    ON)
-
+option(BUILD_TESTS      "Build deepworks with tests"         ON)
+option(WITH_EIGEN       "Build deepworks with eigen backend" ON)
+option(BUILD_SAMPLES    "Build samples"                      ON)
+option(BUILD_BENCHMARKS "Build benchmarks"                   ON)
+option(DOWNLOAD_DATA    "Download and unpack datasets"       ON)
```
Comment (l-bat) [inline CMakeLists.txt:26]: Please add  `DOWNLOAD_DATA` to README.md

File: samples/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+## DeepWorks samples
+
+Before using scripts setup enviroment:
+
+* Build deepworks with samples:
+```bash
+mkdir build
+cd build
+cmake <path-to-deepworks-root> -DBUILD_SAMPLES=ON -DDOWNLOAD_DATA=ON
+make -j8
```
Comment (l-bat) [inline samples/README.md:10]: I think there is no need to duplicate the build process in samples/README.md.

Comment (TolyaTalamanov) [reply samples/README.md]: Done

File: CMakeLists.txt (hunk 3 above)
Comment (TolyaTalamanov) [reply CMakeLists.txt:26]: Done

**Review by l-bat — APPROVED** [review]
Thanks 👍

**Pull request merged**